alembic/versions/472c199d7d44_set_nullable_false_on_state_device.py

```diff
@@ -0,0 +1,32 @@
+"""set nullable false on state device
+
+Revision ID: 472c199d7d44
+Revises: af8e4f84b552
+Create Date: 2019-07-22 12:39:21.595504
+
+"""
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+
+# revision identifiers, used by Alembic.
+revision = '472c199d7d44'
+down_revision = 'af8e4f84b552'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.alter_column('devices', 'state',
+               existing_type=postgresql.ENUM('active', 'archived', 'deleted', name='statetype'),
```

Comment:
For this PR to solve the problem at hand, we must set all the existing devices to a default `active`. To do that, add a comma and `server_default='active'` after `nullable=True`. This will solve the problem at hand such that when the migrations are done, all existing devices will be set to active.

Reply:
Thanks for the clarification @michael-basweti.

Reply:
Thank you @michael-basweti this now works perfectly.